### PR TITLE
JENKINS-42651 add support for running single functional test on ALM server

### PR DESCRIPTION
**Pull request by jackzenko**
Jenkins does not support running single test on ALM server, if user type a path points to test rather than test set, Jenkins will return "Test folder not found on ALM" error.
Added support for running single test on ALM.

Note:
This is a customer escalation by Nike Inc, the hot fix was verified by Nike Inc without any issue. Benny requested to add this change to our repository so other customer would be able to run single test.

 JIRA ticket link:
https://issues.jenkins-ci.org/browse/JENKINS-42651
Corresponding AGM user story:
24143 Nike Jenkins CPE Support for running single test

### CODE REVIEW

**Comment by jackzenko** [comment]
tsName is supposed to be changed at this point, there is no point to introduce a new variable.
please ignore this sonar issue

changing code from:
a = 10;

to:
int b;
b = 10;
a = b;

does not make it any better to read.

**Comment by YafimK** [comment]
- Not only that it's better to read - it's also less error prone. 
- Changing method parameters could lead to future errors and to less readable code - instead you should use the second variable as the intermediary variable all along the method and to leave the method parameter intact.

**more on that here:**
https://dev.eclipse.org/sonar/rules/show/squid:S1226
http://stackoverflow.com/questions/13507025/avoid-reassigning-parameters

- Moreover, it's suggested to use clean variable names that hint on their use - "tName" doesn't. it's not critical.

**Waiting for your fix to merge.**

File: HpToolsLauncher/Runners/AlmTestSetsRunner.cs [hunk 1]
```diff
@@ -383,17 +383,41 @@ public TestSuiteRunResults RunTestSet(string tsFolderName, string tsName, double
...
+			// test set not found, try to find specific test by path
+            if(tsFolder == null)
+            {
+                // if test set path was not found, the path may points to specific test
+                // remove the test name and try find test set with parent path
+                try
+                {
+                    int pos = tsPath.LastIndexOf("\\") + 1;
+                    tName = tsName;
+                    tsName = tsPath.Substring(pos, tsPath.Length - pos);
```
Comment (YafimK) [inline HpToolsLauncher/Runners/AlmTestSetsRunner.cs]: Method parameter "tsName" is reassigned here.

**Comment by jackzenko** [comment]
sonar fix committed

**Comment by YafimK** [comment]
Approved for test build

**Comment by YafimK** [comment]
Ignore codacy - seems irrelevant issue.

**Pull request merged**